packager/VisibilityChecker.cc

```diff
@@ -510,8 +538,19 @@ class VisibilityCheckerPass final {
                     auto pkgName = gs.packageDB().getPackageNameForFile(f.file);
                     if (pkgName.exists()) {
                         core::Context ctx{gs, core::Symbols::root(), f.file};
-                        VisibilityCheckerPass pass{ctx, gs.packageDB().getPackageInfo(pkgName)};
+                        const auto &package = gs.packageDB().getPackageInfo(pkgName);
+                        VisibilityCheckerPass pass{ctx, package};
                         ast::TreeWalk::apply(ctx, pass, f.tree);
+                        for (auto &[symbol, locs] : pass.imports) {
+                            fast_sort(locs,
+                                      [](const auto a, const auto b) -> bool { return a.beginPos() > b.beginPos(); });
+                            reportMissingImport(ctx, package, symbol, locs);
+                        }
+                        for (auto &[symbol, locs] : pass.testImports) {
+                            fast_sort(locs,
+                                      [](const auto a, const auto b) -> bool { return a.beginPos() > b.beginPos(); });
+                            reportMissingTestImport(ctx, package, symbol, locs);
+                        }
```

Comment:
This only deduplicates within a single file.

If you're missing an import for the symbol and have two references in separate files, Sorbet will still generate two imports.

Reply:
Fixed. I've also added a test with two references to a class in the same file and two references to a class in different files